yogstation/code/datums/components/crawl.dm

```diff
@@ -381,9 +385,32 @@
 		for(var/obj/item/vomitcrawl/B in C)
 			qdel(B)
 	..()
+	crawl.holder = null
+	qdel(crawl)
+	crawl = null
 	user.visible_message("<span class='warning'><B>[user] rises out of the pool of vomit!?</B></span>")
 	exit_vomit_effect(target, user)
 
+/obj/item/vomitcrawlholder	//if the pool of vomit we entered gets destroyed, we appear stunned on top of it
+	desc = "If you see this, contact a coder"
+	var/datum/component/crawl/vomit/holder
+	var/mob/living/mob
+	
+/obj/item/vomitcrawlholder/Destroy()
+	if(holder)
+		if(iscarbon(mob))
+			var/mob/living/carbon/C = mob
+			for(var/obj/item/vomitcrawl/B in C)
+				qdel(B)
```

Comment:
Not a major thing or anything, but why B? The convention is to use a letter relevant, like V for vomit. Not a big deal tho

Reply:
Oh, i copied bloodcrawl and B standed for that, might as well change it i guess.